### PR TITLE
Reindex protected dossiers when blocking local roles

### LINKED ISSUE #3650: Admin View "Geschützte Objekte" does not list protected dossiers
Dossiers protected via the new fields IProtectDossier are not listed in the "Geschützte Objekte" Admin view.

Steps to reproduce:
1. Add a new dossier.
2. Select a group or a user in the new field "Lesend & Schreibend"
3. Save 
--> The dossier is not listed in the admin view.

### PR DESCRIPTION
**Pull request by Rotonen**
Simply a case of not reindexing when setting the blocked local roles attribute.

Trivial fix on a fresh unreleased feature -> no changelog entry.

Closes #3650

### CODE REVIEW

File: opengever/sharing/tests/test_blocked_local_roles_listing.py [hunk 1]
```diff
@@ -319,3 +320,59 @@ def test_blocked_role_tab_can_drill_down_to_repofolder(self, browser):
...
+        browser.fill({'Title': 'My Dossier'})
+        form = browser.find_form_by_field('Reading')
+        form.find_widget('Reading and writing').fill(self.regular_user.getId())
+        browser.click_on('Save')
+        new_dossier = browser.context
+
+        browser.append_request_header('Accept-Language', 'de-ch')
+        self.login(self.manager, browser)
+
+        self.dossier.__ac_local_roles_block__ = True
```
Comment (phgross) [inline opengever/sharing/tests/test_blocked_local_roles_listing.py]: Should not be necessary the roles should already be blocked and reindexed

File: opengever/sharing/tests/test_blocked_local_roles_listing.py [hunk 2]
```diff
@@ -319,3 +320,59 @@ def test_blocked_role_tab_can_drill_down_to_repofolder(self, browser):
...
+        browser.fill({'Title': 'My Dossier'})
+        form = browser.find_form_by_field('Reading')
+        form.find_widget('Reading and writing').fill(self.regular_user.getId())
+        browser.click_on('Save')
+        new_dossier = browser.context
+
+        browser.append_request_header('Accept-Language', 'de-ch')
+        self.login(self.administrator, browser)
+
+        self.dossier.__ac_local_roles_block__ = True
```
Comment (phgross) [inline opengever/sharing/tests/test_blocked_local_roles_listing.py]: Same here ... Should not be necessary the roles should already be blocked and reindexed

File: opengever/sharing/tests/test_blocked_local_roles_listing.py (hunk 1 above)
Comment (Rotonen) [reply opengever/sharing/tests/test_blocked_local_roles_listing.py]: Copypaste fail from the other tests. Sorry.

**Comment by Rotonen** [comment]
Ready for rereview.

**Pull request merged**